File: android_java/demo_menu/app/src/main/java/com/heyangpeng/demo_menu/SecondActivity.java
package com.heyangpeng.demo_menu;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

public class SecondActivity extends MainActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);
        addFragment(new FirstFragment());
    }

    private void addFragment(FirstFragment firstFragment) {
        FragmentManager fragmentManager = getSupportFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        fragmentTransaction.replace(R.id.second_fragment, firstFragment);
        fragmentTransaction.commit();
    }


    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        super.onCreateOptionsMenu(menu);//加载父类的menu
        getMenuInflater().inflate(R.menu.menu_second, menu);//加载自己的menu
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {

        switch (item.getItemId()) {
            case R.id.add_item_second:
                Toast.makeText(this, "ADD_SECOND", Toast.LENGTH_SHORT).show();
                return true;
            case R.id.remove_item_second:
                Toast.makeText(this, "REMOVE_SECOND", Toast.LENGTH_SHORT).show();
                return true;
            default:
                return super.onOptionsItemSelected(item);

        }
    }

}

File: android_kotlin/menu_demo/app/src/main/java/com/heyangpeng/menu_demo/SecondActivity.kt
package com.heyangpeng.menu_demo

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.Toast
import androidx.fragment.app.Fragment

class SecondActivity : MainActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)
        addFragment(FirstFragment())
    }

    private fun addFragment(fragment: Fragment) {
        val fragmentManager = supportFragmentManager
//        val transition = fragmentManager.beginTransaction()
//        transition.replace(R.id.second_fragment, fragment)
//        transition.commit()
        fragmentManager.beginTransaction().apply {
            replace(R.id.second_fragment, fragment)
            commit()
        }
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        super.onCreateOptionsMenu(menu)
        menuInflater.inflate(R.menu.menu_second, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.add_item_second -> Toast.makeText(this, "ADD_SECOND", Toast.LENGTH_SHORT).show()
            R.id.remove_item_second -> Toast.makeText(this, "REMOVE_SECOND", Toast.LENGTH_SHORT)
                .show()
        }
        return super.onOptionsItemSelected(item)
    }
}